lib/types/index.d.ts

```diff
@@ -599,6 +599,29 @@ export namespace Rule {
 		create(context: RuleContext): NodeListener;
 	}
 
+	type JSRuleDefinitionTypeOptions = {
+		RuleOptions: unknown[];
+		MessageIds: string;
+		ExtRuleDocs: Record<string, unknown>;
+	};
+
+	type JSRuleDefinition<
```

Comment:
If we are going to expose this, I think it should be outside of any namespace to match how plugins work. @fasttime what do you think?

Reply:
I thought `Rule.JSRuleDefinition` was okay, but sounds good to export `JSRuleDefinition` outside a namespace for consistency 👍